src/components/Warehouse/WarehouseCard.js

```diff
@@ -3,14 +3,80 @@ import RightChevron from "../../assets/icons/chevron_right-24px.svg";
 import DeleteIcon from "../../assets/icons/delete_outline-24px.svg";
 import EditIcon from "../../assets/icons/edit-24px.svg";
 import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import DeleteWarehouse from "../Dialog/DeleteWarehouse";
+import { useRef, useState, useEffect } from "react";
 
 const WarehouseCard = ({ item }) => {
   const navigate = useNavigate();
+  const [warehouses, setWarehouses] = useState([]);
+  const id = item.id;
+  const idWarehouseRef = useRef();
+  const [dialog, setDialog] = useState({
+    warehouseName: "",
+    isLoading: false,
+  });
 
   const handleWarehouseClick = (id) => {
     navigate(`/warehouses/${id}/inventories`);
   };
 
+  useEffect(() => {
+    const fetchWarehouse = async () => {
+      try {
+        const resp = await axios.get(
+          `http://localhost:8080/api/warehouses/${id}`
+        );
+        setWarehouses(resp.data);
+      } catch (error) {
+        console.error("Error fetching a warehouse data:", error);
+      }
+    };
+    fetchWarehouse();
+  }, [id]);
+
+  const handleDialog = (warehouseName, isLoading) => {
+    setDialog({
+      warehouseName,
+      isLoading,
+    });
+  };
+
+  const deleteWarehouse = async () => {
+    let response = null;
+    try {
+      response = await axios.delete(
+        `http://localhost:8080/api/warehouses/${id}`
+      );
+    } catch (err) {
+      console.error(err);
+    }
+    if (response.data) {
+      return response.data;
+    } else {
+      return null;
+    }
+  };
+
+  const handleDeleteWarehouse = (e, id) => {
+    e.preventDefault();
+
+    handleDialog(warehouses.warehouse_name, true);
+
+    idWarehouseRef.current = id;
+
+    setWarehouses(warehouses);
+  };
+
+  const areUSureUWantToDelete = (choose) => {
+    if (choose) {
+      deleteWarehouse();
+      window.location.reload();
```

Comment:
Loading an entire page when it should be a state update is not a good practice in React. 

What we should instead do is on successful delete, use array.filter to filter the warehouse items by deleted id and update that state instead of reloading the page.